src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpPossiblyDeclareAsNullableDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(s_possiblyDeclareAsNullableRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression,
+                SyntaxKind.IsPatternExpression, SyntaxKind.ConditionalAccessExpression);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var fixableSymbol = IsFixable(context.Node, context.SemanticModel);
+            if (fixableSymbol != null)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(s_possiblyDeclareAsNullableRule, context.Node.GetLocation(), fixableSymbol.Name));
+            }
+        }
+
+        internal static ISymbol IsFixable(SyntaxNode node, SemanticModel model)
+        {
+            var symbolToFix = TryGetSymbolToFix(node, model);
+            if (symbolToFix == null ||
+                symbolToFix.Locations.Length != 1 ||
+                !symbolToFix.IsNonImplicitAndFromSource() ||
+                symbolToFix.Locations[0].SourceTree != node.SyntaxTree)
+            {
+                return null;
+            }
+
+            if (!IsFixableType(symbolToFix))
+            {
+                return null;
+            }
+
+            return symbolToFix;
+        }
+
+        private static bool IsFixableType(ISymbol symbolToFix)
+        {
+            var type = symbolToFix switch
+            {
+                IParameterSymbol parameter => parameter.Type,
+                ILocalSymbol local => local.Type,
+                IPropertySymbol property => property.Type,
+                IMethodSymbol method when method.IsDefinition => method.ReturnType,
+                IFieldSymbol field => field.Type,
+                _ => null
+            };
+
+            return type?.IsReferenceType == true;
+        }
+
+        private static ISymbol TryGetSymbolToFix(SyntaxNode node, SemanticModel model)
+        {
+            ExpressionSyntax value = null;
+            if (node.IsKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression))
+            {
+                var equals = (BinaryExpressionSyntax)node;
+                if (equals.Right.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    value = equals.Left;
+                }
+                else if (equals.Left.IsKind(SyntaxKind.NullLiteralExpression))
+                {
+                    value = equals.Right;
+                }
+            }
+
+            if (node.IsKind(SyntaxKind.IsPatternExpression))
```

Comment:
`.IsKind(..., out IsPatternExpression patternExpr)`. (and the rest of the checks)